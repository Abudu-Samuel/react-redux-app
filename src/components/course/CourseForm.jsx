import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../common/TextInput';
import SelectInput from '../common/SelectInput';

const CourseForm = ({
  course, allAuthors, onSave, onChange, errors, isSaving, courseToModify
}) => (
  <form>
    <TextInput
        name="title"
        label="Title"
        defaultValue={courseToModify.title}
        onChange={onChange}
        placeholder="Title"
        error={errors.title}
      />
    <SelectInput
    name="authorId"
    label="Author"
    defaultValue={courseToModify.authorId}
    defaultOption="Select Author"
    options={allAuthors}
    onChange={onChange}
    error={errors.authorId}
    />

    <TextInput
      name="category"
      label="Category"
      placeholder="Category"
      defaultValue={courseToModify.category}
      onChange={onChange}
      error={errors.category}
    />

    <TextInput
      name="length"
      label="Length"
      placeholder="Length"
      onChange={onChange}
      defaultValue={courseToModify.length}
      error={errors.length}
    />

    <input
      name="submit"
      disabled={isSaving}
      defaultValue={isSaving ? 'saving' : 'Save'}
      className="btn btn-sm btn-primary"
      onClick={onSave}
    />
  </form>
);

CourseForm.propTypes = {
  course: PropTypes.object.isRequired,
  allAuthors: PropTypes.array.isRequired,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  isSaving: PropTypes.bool.isRequired,
  errors: PropTypes.object.isRequired
};

export default CourseForm;