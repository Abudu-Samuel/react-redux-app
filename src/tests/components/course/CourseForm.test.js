/* eslint-disable */
import CourseForm from '../../../components/course/CourseForm';

const config = (isSaving = false) => {
  const props = {
    allAuthors: [{}],
    onSave: jest.fn(),
    onChange: jest.fn(),
    errors: {},
    isSaving,
    courseToModify: [{}]
  };
  return {
    wrap: mount(<CourseForm {...props} />),
    props
  }
};

describe('<CourseForm />', () => {
  it('should renderCourseForm component', () => {
    const { wrap } = config();
    expect(toJson(wrap)).toMatchSnapshot();
  });

  it('should renderCourseForm component', () => {
    const { wrap } = config(true);
    expect(wrap.find('input').at(3).props().defaultValue).toBe('saving');
  });
});