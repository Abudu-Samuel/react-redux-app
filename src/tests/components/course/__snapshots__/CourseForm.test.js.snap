// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseForm /> should renderCourseForm component 1`] = `
<CourseForm
  allAuthors={
    Array [
      Object {},
    ]
  }
  courseToModify={
    Array [
      Object {},
    ]
  }
  errors={Object {}}
  isSaving={false}
  onChange={[MockFunction]}
  onSave={[MockFunction]}
>
  <form>
    <TextInput
      label="Title"
      name="title"
      onChange={[MockFunction]}
      placeholder="Title"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="title"
        >
          Title
        </label>
        <div
          className="field"
        >
          <input
            className="form-control"
            name="title"
            onChange={[MockFunction]}
            placeholder="Title"
            type="text"
          />
        </div>
      </div>
    </TextInput>
    <SelectInput
      defaultOption="Select Author"
      label="Author"
      name="authorId"
      onChange={[MockFunction]}
      options={
        Array [
          Object {},
        ]
      }
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="authorId"
        >
          Author
        </label>
        <div
          className="field"
        >
          <select
            className="form-control"
            name="authorId"
            onChange={[MockFunction]}
          >
            <option
              value=""
            >
              Select Author
            </option>
            <option />
          </select>
        </div>
      </div>
    </SelectInput>
    <TextInput
      label="Category"
      name="category"
      onChange={[MockFunction]}
      placeholder="Category"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <div
          className="field"
        >
          <input
            className="form-control"
            name="category"
            onChange={[MockFunction]}
            placeholder="Category"
            type="text"
          />
        </div>
      </div>
    </TextInput>
    <TextInput
      defaultValue={1}
      label="Length"
      name="length"
      onChange={[MockFunction]}
      placeholder="Length"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="length"
        >
          Length
        </label>
        <div
          className="field"
        >
          <input
            className="form-control"
            defaultValue={1}
            name="length"
            onChange={[MockFunction]}
            placeholder="Length"
            type="text"
          />
        </div>
      </div>
    </TextInput>
    <input
      className="btn btn-sm btn-primary"
      defaultValue="Save"
      disabled={false}
      name="submit"
      onClick={[MockFunction]}
    />
  </form>
</CourseForm>
`;
